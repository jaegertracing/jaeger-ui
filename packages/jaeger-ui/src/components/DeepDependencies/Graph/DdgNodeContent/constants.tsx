// Copyright (c) 2019 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { getUrl as getSearchUrl } from '../../../SearchTracePage/url';

// Constants for calculating circular node positioning
export const FONT_SIZE = 14;
export const FONT = `${FONT_SIZE}px Helvetica Nueue`;
export const LINE_HEIGHT = 1.5;
export const OP_PADDING_TOP = 5;
export const PROGRESS_BAR_STROKE_WIDTH = 15;
export const RADIUS = 75;
export const WORD_RX = /\W*\w+\W*/g;

// While browsers suport URLs of unlimited length, many server clients do not handle more than this max
// export const MAX_LENGTH = 2083;
// get unpreditable, unreproducable results past 7400 characters
export const MAX_LENGTH = 7000;
export const MAX_LINKED_TRACES = 35;
export const MIN_LENGTH = getSearchUrl().length;
export const PARAM_NAME_LENGTH = '&traceID='.length;
