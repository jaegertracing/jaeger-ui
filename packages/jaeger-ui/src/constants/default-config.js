// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

export const FALLBACK_DAG_MAX_NUM_SERVICES = 100;
export const FALLBACK_TRACE_NAME_REGEXP_ALLOWLIST = [];

export const deprecations = [
  {
    formerKey: 'dependencies.dagMaxNumServices',
    currentKey: 'dependencies.menuEnabled',
    formerValue: 100,
    currentValue: false,
    callback: (config, formerKey, currentKey, formerValue, currentValue) => {
      const dagMaxNumServices = config.dependencies && config.dependencies.dagMaxNumServices;
      if (dagMaxNumServices !== FALLBACK_DAG_MAX_NUM_SERVICES) {
        // set the current key to true because the former value was not the default
        if (config.dependencies) {
          config.dependencies.menuEnabled = true;
        } else {
          config.dependencies = { menuEnabled: true };
        }
        // eslint-disable-next-line no-console
        console.warn(
          `Deprecated configuration key (${formerKey}: ${dagMaxNumServices}) detected, ` +
            `please use (${currentKey}: true) instead.`
        );
      }
    },
  },
];

export default {
  // Setting archiveEnabled to null means:
  // 1. Respect backend capability for non-memory storage
  // 2. Disable for in-memory storage (handled in getArchivedEnabled function)
  archiveEnabled: null,
  dependencies: {
    dagMaxNumServices: FALLBACK_DAG_MAX_NUM_SERVICES,
    menuEnabled: true,
  },
  search: {
    maxLookback: {
      label: '2 Days',
      value: '2d',
    },
    maxLimit: 1500,
  },
  tracking: {
    gaID: null,
    trackErrors: true,
    customWebAnalytics: null,
  },
  linkPatterns: [],
  menu: [
    {
      label: 'About Jaeger',
      items: [
        {
          label: 'GitHub',
          url: 'https://github.com/jaegertracing/jaeger',
        },
        {
          label: 'Docs',
          url: 'https://www.jaegertracing.io/docs/latest',
        },
        {
          label: 'Twitter',
          url: 'https://twitter.com/JaegerTracing',
        },
        {
          label: 'Discussion Group',
          url: 'https://groups.google.com/forum/#!forum/jaeger-tracing',
        },
        {
          label: 'Gitter.im',
          url: 'https://gitter.im/jaegertracing/Lobby',
        },
        {
          label: 'Blog',
          url: 'https://medium.com/jaegertracing',
        },
      ],
    },
  ],
  traceNameRegExpAllowList: FALLBACK_TRACE_NAME_REGEXP_ALLOWLIST,
};