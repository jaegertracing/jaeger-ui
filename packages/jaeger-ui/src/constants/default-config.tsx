// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import deepFreeze from 'deep-freeze';

import { FALLBACK_DAG_MAX_NUM_SERVICES } from './index';

export default deepFreeze(
  Object.defineProperty(
    {
      archiveEnabled: false,
      dependencies: {
        dagMaxNumServices: FALLBACK_DAG_MAX_NUM_SERVICES,
        menuEnabled: true,
      },
      linkPatterns: [],
      menu: [
        {
          label: 'About Jaeger',
          items: [
            {
              label: 'GitHub',
              url: 'https://github.com/uber/jaeger',
            },
            {
              label: 'Docs',
              url: 'http://jaeger.readthedocs.io/en/latest/',
            },
            {
              label: 'Twitter',
              url: 'https://twitter.com/JaegerTracing',
            },
            {
              label: 'Discussion Group',
              url: 'https://groups.google.com/forum/#!forum/jaeger-tracing',
            },
            {
              label: 'Gitter.im',
              url: 'https://gitter.im/jaegertracing/Lobby',
            },
            {
              label: 'Blog',
              url: 'https://medium.com/jaegertracing/',
            },
          ],
        },
      ],
      search: {
        maxLookback: {
          label: '2 Days',
          value: '2d',
        },
        maxLimit: 1500,
      },
      tracking: {
        gaID: null,
        trackErrors: true,
      },
    },
    // fields that should be individually merged vs wholesale replaced
    '__mergeFields',
    { value: ['dependencies', 'search', 'tracking'] }
  )
);

export const deprecations = [
  {
    formerKey: 'dependenciesMenuEnabled',
    currentKey: 'dependencies.menuEnabled',
  },
  {
    formerKey: 'gaTrackingID',
    currentKey: 'tracking.gaID',
  },
];
