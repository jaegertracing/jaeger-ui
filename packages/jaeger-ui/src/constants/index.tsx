// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

export const TOP_NAV_HEIGHT = 47 as 47;

export const FALLBACK_DAG_MAX_NUM_SERVICES = 100 as 100;
export const FALLBACK_TRACE_NAME = '<trace-without-root-span>' as '<trace-without-root-span>';
export const FALLBACK_STANDARD_DATETIME_FORMAT = 'LLL';

export const FETCH_DONE = 'FETCH_DONE' as 'FETCH_DONE';
export const FETCH_ERROR = 'FETCH_ERROR' as 'FETCH_ERROR';
export const FETCH_LOADING = 'FETCH_LOADING' as 'FETCH_LOADING';

export const fetchedState = {
  DONE: FETCH_DONE,
  ERROR: FETCH_ERROR,
  LOADING: FETCH_LOADING,
};
