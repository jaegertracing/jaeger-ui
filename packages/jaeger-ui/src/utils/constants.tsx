// Copyright (c) 2023 The Jaeger Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

/**
 * Provides access to constants injected by the build system.
 */

/**
 * Get the current execution environment, as inferred from NODE_ENV at build time.
 */
export function getAppEnvironment() {
  return __APP_ENVIRONMENT__;
}

/**
 * Get injected version details as a JSON-formatted string.
 */
export function getVersionInfo() {
  return __REACT_APP_VSN_STATE__;
}

export function shouldDebugGoogleAnalytics() {
  return __REACT_APP_GA_DEBUG__;
}
