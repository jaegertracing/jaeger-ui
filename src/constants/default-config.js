// Copyright (c) 2017 Uber Technologies, Inc.
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
// THE SOFTWARE.

export default {
  menu: [
    {
      label: 'About Jaeger',
      items: [
        {
          label: 'GitHub',
          url: 'https://github.com/uber/jaeger',
        },
        {
          label: 'Docs',
          url: 'http://jaeger.readthedocs.io/en/latest/',
        },
        {
          label: 'Twitter',
          url: 'https://twitter.com/JaegerTracing',
        },
        {
          label: 'Discussion Group',
          url: 'https://groups.google.com/forum/#!forum/jaeger-tracing',
        },
        {
          label: 'Gitter.im',
          url: 'https://gitter.im/jaegertracing/Lobby',
        },
        {
          label: 'Blog',
          url: 'https://medium.com/jaegertracing/',
        },
      ],
    },
  ],
};
