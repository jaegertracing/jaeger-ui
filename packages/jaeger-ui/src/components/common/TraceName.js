// @flow

// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';

import BreakableText from '../common/BreakableText';
import LoadingIndicator from '../common/LoadingIndicator';
import { fetchedState, FALLBACK_TRACE_NAME } from '../../constants';

import type { FetchedState } from '../../types';
import type { ApiError } from '../../types/api-error';

import './TraceName.css';

type Props = {
  className?: string,
  error?: ?ApiError,
  state?: ?FetchedState,
  traceName?: ?string,
};

export default function TraceName(props: Props) {
  const { className, error, state, traceName } = props;
  const isErred = state === fetchedState.ERROR;
  let title = traceName || FALLBACK_TRACE_NAME;
  let errorCssClass = '';
  if (isErred) {
    errorCssClass = 'is-error';
    let titleStr = '';
    if (error) {
      titleStr = typeof error === 'string' ? error : error.message || String(error);
    }
    if (!titleStr) {
      titleStr = 'Error: Unknown error';
    }
    title = <BreakableText text={titleStr} />;
  } else if (state === fetchedState.LOADING) {
    title = <LoadingIndicator small />;
  } else {
    const text = traceName || FALLBACK_TRACE_NAME;
    title = <BreakableText text={text} />;
  }
  return <span className={`TraceName ${errorCssClass} ${className || ''}`}>{title}</span>;
}

TraceName.defaultProps = {
  className: '',
  error: null,
  state: null,
  traceName: null,
};
