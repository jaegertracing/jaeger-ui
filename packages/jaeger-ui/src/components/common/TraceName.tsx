// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';

import BreakableText from './BreakableText';
import LoadingIndicator from './LoadingIndicator';
import { fetchedState, FALLBACK_TRACE_NAME } from '../../constants';

import { FetchedState, TNil } from '../../types';
import { ApiError } from '../../types/api-error';

import './TraceName.css';

type Props = {
  className?: string;
  error?: ApiError | TNil;
  state?: FetchedState | TNil;
  traceName?: string | TNil;
};

export default function TraceName(props: Props) {
  const { className, error, state, traceName } = props;
  const isErred = state === fetchedState.ERROR;
  let title: string | React.ReactNode = traceName || FALLBACK_TRACE_NAME;
  let errorCssClass = '';
  if (isErred) {
    errorCssClass = 'is-error';
    let titleStr = '';
    if (error) {
      titleStr = typeof error === 'string' ? error : error.message || String(error);
    }
    if (!titleStr) {
      titleStr = 'Error: Unknown error';
    }
    title = titleStr;
    title = <BreakableText text={titleStr} />;
  } else if (state === fetchedState.LOADING) {
    title = <LoadingIndicator small />;
  } else {
    const text = String(traceName || FALLBACK_TRACE_NAME);
    title = <BreakableText text={text} />;
  }
  return <span className={`TraceName ${errorCssClass} ${className || ''}`}>{title}</span>;
}
