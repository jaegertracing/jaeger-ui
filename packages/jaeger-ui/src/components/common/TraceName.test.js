// Copyright (c) 2020 The Jaeger Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import { shallow } from 'enzyme';
import TraceName from './TraceName';
import { fetchedState } from '../../constants';

describe('<TraceName>', () => {
  let wrapper;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {};
    wrapper = shallow(<TraceName {...defaultProps} />);
  });

  it('renders with default props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with className', () => {
    const props = { ...defaultProps, className: 'TEST-CLASS-NAME' };
    wrapper = shallow(<TraceName {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders with traceName', () => {
    const props = { ...defaultProps, traceName: 'TEST-TRACE-NAME' };
    wrapper = shallow(<TraceName {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders in loading state', () => {
    const props = { ...defaultProps, state: fetchedState.LOADING };
    wrapper = shallow(<TraceName {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders in error state', () => {
    const props = { ...defaultProps, state: fetchedState.ERROR, error: 'TEST-ERROR-MESSAGE' };
    wrapper = shallow(<TraceName {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders error object in error state', () => {
    const props = { ...defaultProps, state: fetchedState.ERROR, error: new Error('ERROR-OBJECT-MESSAGE') };
    wrapper = shallow(<TraceName {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders empty string error in error state', () => {
    const props = { ...defaultProps, state: fetchedState.ERROR, error: '' };
    wrapper = shallow(<TraceName {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders error object with empty message in error state', () => {
    const props = { ...defaultProps, state: fetchedState.ERROR, error: new Error('') };
    wrapper = shallow(<TraceName {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
