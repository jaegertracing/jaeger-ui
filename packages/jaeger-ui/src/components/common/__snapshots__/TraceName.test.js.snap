// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TraceName> renders empty string error in error state 1`] = `
<span
  className="TraceName is-error "
>
  <BreakableText
    className="BreakableText"
    text="Error: Unknown error"
    wordRegexp={/\\\\W\\*\\\\w\\+\\\\W\\*/g}
  />
</span>
`;

exports[`<TraceName> renders error object in error state 1`] = `
<span
  className="TraceName is-error "
>
  <BreakableText
    className="BreakableText"
    text="ERROR-OBJECT-MESSAGE"
    wordRegexp={/\\\\W\\*\\\\w\\+\\\\W\\*/g}
  />
</span>
`;

exports[`<TraceName> renders error object with empty message in error state 1`] = `
<span
  className="TraceName is-error "
>
  <BreakableText
    className="BreakableText"
    text="Error"
    wordRegexp={/\\\\W\\*\\\\w\\+\\\\W\\*/g}
  />
</span>
`;

exports[`<TraceName> renders in error state 1`] = `
<span
  className="TraceName is-error "
>
  <BreakableText
    className="BreakableText"
    text="TEST-ERROR-MESSAGE"
    wordRegexp={/\\\\W\\*\\\\w\\+\\\\W\\*/g}
  />
</span>
`;

exports[`<TraceName> renders in loading state 1`] = `
<span
  className="TraceName  "
>
  <LoadingIndicator
    centered={false}
    small={true}
  />
</span>
`;

exports[`<TraceName> renders with className 1`] = `
<span
  className="TraceName  TEST-CLASS-NAME"
>
  <BreakableText
    className="BreakableText"
    text="<trace-without-root-span>"
    wordRegexp={/\\\\W\\*\\\\w\\+\\\\W\\*/g}
  />
</span>
`;

exports[`<TraceName> renders with default props 1`] = `
<span
  className="TraceName  "
>
  <BreakableText
    className="BreakableText"
    text="<trace-without-root-span>"
    wordRegexp={/\\\\W\\*\\\\w\\+\\\\W\\*/g}
  />
</span>
`;

exports[`<TraceName> renders with traceName 1`] = `
<span
  className="TraceName  "
>
  <BreakableText
    className="BreakableText"
    text="TEST-TRACE-NAME"
    wordRegexp={/\\\\W\\*\\\\w\\+\\\\W\\*/g}
  />
</span>
`;
